### ISSUE TITLE
Jade to Pug

### ISSUE THREAD

**Issue by genoma** [devongovett/atom-jade#63]
About the pull request https://github.com/devongovett/atom-jade/pull/62

I think we need to change the name of the library to something like **language-pug**, just to update to the new name and to bring some consistency between the repository and the name of the plugin.

Let me know what do you thing about this.

**Comment by vizo** [comment]
:+1:

**Comment by devongovett** [comment]
Wow, crazy. https://github.com/pugjs/pug/issues/2184 for context.

**Comment by devongovett** [comment]
Maybe publish both language-jade and language-pug based on this package somehow so that they both work? @LinusU what do you think?

**Comment by LinusU** [comment]
Hmm, yeah we should absolutely change the name to `language-pug`, and I guess it would be nice to publish it under both names.

I don't know how much job that is though.

I wonder if there is a way to publish an update that somehow sends a message to the user that we have changed name...

**Comment by vizo** [comment]
@genoma already made language-pug ... he wrote it on feb now is may :)

**Comment by RobertWHurst** [comment]
I just wanted to add, this is a better implementation than the current language-pug package. That package has some annoying bugs when it comes to syntax highlighting. For example it doesn't work correctly with inline elements, or with code blocks (with the period syntax). It also does not offer the same level of configuration. Any chance we can add support for the pug extension to this package, or perhaps publish it under another name for pug anyway?

**Comment by RobertWHurst** [comment]
I just wanted to add, this is a better implementation than the current language-pug package. That package has some annoying bugs when it comes to syntax highlighting. For example it doesn't work correctly with inline elements, or with code blocks (with the period syntax). It also does not offer the same level of configuration. Any chance we can add support for the pug extension to this package, or perhaps publish it under another name for pug anyway?

**Comment by RobertWHurst** [comment]
I just wanted to add, this is a better implementation than the current language-pug package. That package has some annoying bugs when it comes to syntax highlighting. For example it doesn't work correctly with inline elements, or with code blocks (with the period syntax). It also does not offer the same level of configuration. @devongovett, any chance we can add support for the pug extension to this package, or perhaps publish it under another name for pug anyway?

| language-pug | language-jade |
| :-----------:  | :------------: |
| <img width="461" alt="screen shot 2016-12-26 at 12 53 44 pm" src="https://cloud.githubusercontent.com/assets/178857/21486487/6120863e-cb6a-11e6-8f24-1f5a70014d22.png"> | <img width="462" alt="screen shot 2016-12-26 at 12 54 35 pm" src="https://cloud.githubusercontent.com/assets/178857/21486496/7f9b3b72-cb6a-11e6-9123-5ff771d76ecb.png"> |

**Comment by sethmcleod** [comment]
@RobertWHurst I just published a package called `language-pug-jade` which is a blatant copy of this repo with a few updates so that it supports pug files as well: https://atom.io/packages/language-pug-jade

**Comment by RobertWHurst** [comment]
@sethmcleod Very cool. Thanks for that 🍻

**Comment by in-in** [comment]
You can get support for _.pug_ file if you add that to the `core` section in your `config.cson`
```
customFileTypes:
      'source.jade': [
        'pug'
      ]
```
http://flight-manual.atom.io/using-atom/sections/basic-customization/#customizing-language-recognition

**Comment by LinusU** [comment]
A pull request that adds `pug` as a default extension would be very welcome :)

**Comment by hollenberry** [comment]
> You can get support for .pug file if you add that to the core section in your config.cson

@in-in Thank you!! This might be worth adding to the README.md for people new to Jade/Pug, because I was digging all over the place to learn to set the customFileType correctly, and your comment was the key I needed. 👍

**Comment by qucode1** [comment]
I agree, thanks a lot @in-in. I tested every package I could find for atom, but none worked correctly. Surprisingly I couldn't even find anything useful on google. I thought there would be a lot more people having a highligting issue with pug in atom.

**Comment by williamschneider** [comment]
@RobertWHurst I see in your post from 12-26-16 you had a photo showing some differences. In your photo of "language-jade"  your code after the pound sign had some highlighting that was not in your "language-pug" photo. My current setups using these different packages has a lack of highlighting for code after my '#' signs. Is there a tip you could offer that I am missing? 

-I am new to using Atom and have not worked in the code base for it.

Could this also be a function of the theme I am using? Currently monokai.

Please any help you could give would be great. I want to get the rest of my team on board with Atom but they currently prefer Sublime. Thank you!
